rehive/api/resources/base_resources.py

```diff
@@ -23,48 +25,84 @@ def post(self, data=None, resource_id=None, idempotent_key=None, timeout=None, *
         # Allow us to parse through arbitrary request arguments
         if data is None:
             data = {}
-        request_kwargs = {}
-        if 'file' in kwargs:
-            request_kwargs['files'] = {
-                'file': (open(kwargs.get('file'), 'rb'))
-            }
-            kwargs.pop('file', None)
-        # We need this flag to force non-json on file uploads
-        json = True
-        if 'json' in kwargs:
-            json = kwargs.get('json')
-            kwargs.pop('json', None)
+
         data = {**data, **kwargs}
+        # Get the files.
+        files = {
```

Comment:
**Firstly**: Doing these in 2 dict comprehensions (line 31 and 36) is not very efficient, theoretically you could do this in a single for loop (where you add to `files` and remove from data in the same iteration). Or if you want to do this in the fancy way you should use `map` and have a function that performs the action of adding and removing from `files` and `data` respectively.

**Secondly**:  You repeat lines 30-43 in exactly the same way on lines 58-71 and 85-98. You should attempt to not repeat pieces of self contained code like this when they could instead be moved into a method on the lass or a utility function. In the current form this will be annoying to maintain.

Reply:
During the iteration can't remove or add new data to the same dict. If we need to do this in one loop, then we should `deepcopy` the  dict which is also doing two loops. 

Yea, This peace of code should be moved to a method to remove the repetition.

Reply:
Here is an example of working for loop that iterates through the kwargs only once:
```python
data = {}
files = {}
url = self._build_url(resource_id)
for k, v in kwargs.items():
	if isinstance(v, BufferedReader):
		files[k] = v
	else:
		data[k] = v
```